flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/handlers/JarDeleteHandler.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.webmonitor.handlers;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import org.apache.flink.runtime.instance.ActorGateway;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.io.StringWriter;
+import java.util.Map;
+
+/**
+ * Handles requests for deletion of jars.
+ */
+public class JarDeleteHandler implements RequestHandler, RequestHandler.JsonResponse {
+
+	private final File jarDir;
+
+	public JarDeleteHandler(File jarDirectory) {
+		jarDir = jarDirectory;
+	}
+
+	@Override
+	public String handleRequest(Map<String, String> params, ActorGateway jobManager) throws Exception {
+		final String file = params.get("jarid");
+		try {
+			File[] list = jarDir.listFiles(new FilenameFilter() {
+				@Override
+				public boolean accept(File dir, String name) {
+					return name.equals(file);
+				}
+			});
+			boolean success = false;
+			for (File f: list) {
+				// although next to impossible for multiple files, we still delete them.
+				success = success || f.delete();
+			}
+			StringWriter writer = new StringWriter();
+			JsonGenerator gen = JsonFactory.jacksonFactory.createJsonGenerator(writer);
+			gen.writeStartObject();
+			if (!success) {
+				// this seems to always fail on Windows.
```

Comment:
Yes. I came across that too. I just gave it somewhat lower preference, since we already have too many issues with deletions on Windows. That's the primary reason `mvn verify` fails.
Let me try it out and I will report back if it works.